src/RealtimeClient.ts

```diff
@@ -317,6 +321,76 @@ export default class RealtimeClient {
     })
   }
 
+  /**
+   * Creates a private channel to be used with the provided name.
+   *
+   * @param name Channel name to create
+   */
+  createPrivateChannel(name: string): Promise<string> {
+    const url = `${this.httpEndpoint}/channels?apikey=${this.apiKey}`
```

Comment:
@filipecabaco I think this `apiKey` is supposed to be `service_role` token right? In that case this should be passed as header and not url params.

See: https://github.com/supabase/realtime-js/blob/2798441cbfa6f6dc665b4adcafbfd99d5f8a40f3/src/RealtimeChannel.ts#L439

Reply:
no, we should allow the user to use whatever token he wants as they are able to define if a user is able to create channels.

in the example app for lw11 we will use a service_role token just to make sure people go for the most common approach

Reply:
in any case we should pass any token via `apiKey` header.

Reply:
got it! will change 👍